jaeger-core/src/main/java/io/jaegertracing/propagation/B3TextMapCodec.java

```diff
@@ -80,12 +104,37 @@ public SpanContext extract(TextMap carrier) {
         if (entry.getValue().equals("1")) {
           flags |= DEBUG_FLAG;
         }
+      } else if (entry.getKey().startsWith(baggagePrefix)) {
+        if (baggage == null) {
+          baggage = new HashMap<String, String>();
+        }
+        baggage.put(keys.unprefixedKey(entry.getKey(), baggagePrefix), entry.getValue());
       }
     }
 
     if (null != traceId && null != parentId && null != spanId) {
-      return new SpanContext(traceId, spanId, parentId, flags);
+      SpanContext spanContext = new SpanContext(traceId, spanId, parentId, flags);
+      if (baggage != null) {
+        spanContext = spanContext.withBaggage(baggage);
```

Comment:
this is an interesting use case, what if only baggage is propagated without the rest of the trace context?

Reply:
At the moment it's an invalid span context - null